packages/agents/chain-abstraction/src/helpers/swapdata.ts

```diff
@@ -56,17 +56,21 @@ export const getOriginSwapDataForOneInch = async (args: OriginSwapDataCallbackAr
 /**
  * Returns the `swapData` which will be used on the destination univ2 swapper
  */
-export const getDestinationSwapDataForUniV2 = async (_args: any): Promise<string> => {
+export const getDestinationSwapDataForUniV2 = async (_args: any, path?: any): Promise<string> => {
   const args = _args as UniV2SwapperParams;
-  return defaultAbiCoder.encode(["uint256"], [args.amountOutMin]);
+  return path
+    ? defaultAbiCoder.encode(["uint256", "address[]"], [args.amountOutMin, path])
+    : defaultAbiCoder.encode(["uint256"], [args.amountOutMin]);
 };
 
 /**
  * Returns the `swapData` which will be used on the destination univ3 swapper
  */
-export const getDestinationSwapDataForUniV3 = async (_args: any): Promise<string> => {
+export const getDestinationSwapDataForUniV3 = async (_args: any, path?: any): Promise<string> => {
   const args = _args as UniV3SwapperParams;
-  return defaultAbiCoder.encode(["uint24", "uint256"], [args.poolFee, args.amountOutMin]);
+  if (!path) return defaultAbiCoder.encode(["uint24", "uint256"], [args.poolFee, args.amountOutMin]);
+  const encodedPathData = defaultAbiCoder.encode(["address[]"], [path]);
+  return defaultAbiCoder.encode(["uint24", "uint256", "bytes"], [args.poolFee, args.amountOutMin, encodedPathData]);
 };
```

Comment:
swapdata for univ3 seems not correct.

https://github.com/connext/chain-abstraction-integration/blob/d163e9041ade08c4ef061597511fed926075505b/contracts/shared/Swap/Uniswap/UniV3Swapper.sol#L44
Please check contract code. It requires `(amountMin, path)`.

Reply:
https://github.com/connext/monorepo/commit/9f5df519fee91e5c16045f1e69940d07e5349bcc